dhall/tests/Dhall/Test/Parser.hs

```diff
@@ -108,10 +109,14 @@ notesInLetInLet = do
                             (Just " ")
                             Nothing
                             (Just " ")
-                            (Note "2" (NaturalLit 2))
+                            (Note "2 " (Note "2" (NaturalLit 2)))
```

Comment:
Where do the additional `Note`s come from?

Reply:
The two `Note`s were actually always there, but they used to have the same source span so they would be flattened into one `Note` by this logic:

https://github.com/dhall-lang/dhall-haskell/blob/7eec31d1d72927c4535383534890f180a2b0a85f/dhall/src/Dhall/Parser/Expression.hs#L91-L93

Now that they slightly differ they are no longer flattened in this way

Reply:
I guess we could change that rule to always return the inner `Note` – not sure whether that would be beneficial though…